app/frontend/src/components/ches/HistoryTable.vue

```diff
@@ -57,68 +59,113 @@
+        </v-btn>
+        <v-btn color="primary" small class="ma-2" @click="clearTable">
+          Clear<v-icon small class="ml-2">clear</v-icon>
+        </v-btn>
+      </template>
     </v-data-table>
   </v-container>
 </template>
 
 <script>
 import { format } from 'date-fns';
 import { mapActions, mapGetters } from 'vuex';
+import { mapFields } from 'vuex-map-fields';
 
+import chesService from '@/services/chesService';
 export default {
   name: 'HistoryTable',
   data: () => ({
     error: false,
-    loading: false,
     search: '',
     headers: [
       { text: 'Transaction ID', align: 'start', value: 'txId' },
       { text: 'Message ID', align: 'start', value: 'msgId' },
       { text: 'Tag', align: 'start', value: 'tag' },
       { text: 'Status', align: 'start', value: 'status' },
       { text: 'Delayed', align: 'start', value: 'delayTS', width: 150 },
-      // TODO: implement promote and cancel actions
-      // {
-      //   text: 'Actions',
-      //   align: 'end',
-      //   value: 'actions',
-      //   filterable: false,
-      //   sortable: false,
-      // },
+      { text: 'Actions', align: 'start', value: 'actions', width: 120, filterable: false, sortable: false }
     ],
   }),
 
   computed: {
-    ...mapGetters('ches', ['txIds', 'tableData']),
+    ...mapGetters('ches', ['txIds']),
+    ...mapFields('ches', ['tableData', 'tableLoading'])
   },
 
   methods: {
     ...mapActions('alert', ['showAlert', 'clearAlert']),
-    ...mapActions('ches', ['addTableData', 'populateTable']),
+    ...mapActions('ches', ['populateTable', 'clearTable']),
+
 
-    cancelMessage(msgId) {
-      console.log('cancelMessage', msgId); // eslint-disable-line no-console
+    async cancelMessage(msgId) {
+      try {
+        // wait for OK status from CHES api
+        await chesService.cancel(msgId);
+        // update store with latest table data
+        this.populateTable();
+        // show success alert
+        this.showAlert({
+          type: 'success',
+          text: `<strong>Your message has been successfully cancelled.<br />Message ID: </strong> ${msgId}`,
+        });
+      } catch (e) {
+        this.error = true;
+        // show error alert
+        this.showAlert({
+          type: 'error',
+          text: e,
+        });
+      }
+      window.scrollTo(0, 0);
     },
 
     isTerminalState(status) {
       return ['completed', 'cancelled'].includes(status);
     },
 
-    promoteMessage(msgId) {
-      console.log('promoteMessage', msgId); // eslint-disable-line no-console
+    async promoteMessage(msgId) {
+      try {
+        // wait for OK status from CHES api (does not mean message has been promoted yet)
+        await chesService.promote(msgId);
+        // promote seems to take longer than cancel to complete and show in status from CHES
+        await new Promise(resolve => {
+          setTimeout(resolve, 1000);
+          this.tableLoading = true;
+        });
```

Comment:
The promote request itself will resolve almost instantaneously; however because CHES is queue based, it can take a bit (or a long while in the rare case there is a message backlog) before the message state changes. While a hard-coded 1 second stall will "usually" be fine for most cases, it'd still be better if we can avoid hard-coding a finite time of 1 second into this as this will definitely fail in cases where it takes CHES longer than 1 second to actually dispatch the message and update the message status.